stores/redis/redis.go

```diff
@@ -0,0 +1,344 @@
+	// delete the id-to-url mapping
+	entryKey := entryPathPrefix + id
+	err := r.delValue(entryKey)
+	if err != nil {
+		msg := fmt.Sprintf("Could not delete entry id %s: %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+	// delete the visitors list for the id
+	entryVisitsKey := entryVisitsPrefix + id
+	err = r.delValue(entryVisitsKey)
+	if err != nil {
+		msg := fmt.Sprintf("Could not delete visitors list for id %s: %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// get the user for the id
+	userKey := entryUserPrefix + id
+	var userIdentifier string
+	userIdentifier, err = r.c.Get(userKey).Result()
+	if err != nil {
+		msg := fmt.Sprintf("Could not fetch id to user mapping for id '%s': %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// delete the entry from set of entries for the user
+	userEntriesKey := userToEntriesPrefix + userIdentifier
+	err = r.c.SRem(userEntriesKey, id).Err()
+	if err != nil {
+		msg := fmt.Sprintf("Could not remove entry '%s' from list of entries for user '%s': %v", id, userIdentifier, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	// delete the id-to-user mapping
+	err = r.delValue(userKey)
+	if err != nil {
+		msg := fmt.Sprintf("Could not delete the path-to-user mapping for entry '%s': %v", id, err)
+		logrus.Error(msg)
+		return errors.Wrap(err, msg)
+	}
+
+	return err
+}
+
+func (r *RedisStore) GetEntryByID(id string) (*shared.Entry, error) {
+	var entry *shared.Entry
+	var visitor *shared.Visitor
+	var lastVisit time.Time
+	var raw []byte
+	var err error
+
+	entryKey := entryPathPrefix + id
+	logrus.Debugf("Fetching key: '%s'", entryKey)
+	result := r.c.Get(entryKey)
+	raw, err = result.Bytes()
+	if err != nil {
+		msg := fmt.Sprintf("Error looking up key '%s': %s'", entryKey, err)
+		logrus.Warn(msg)
+		err = shared.ErrNoEntryFound
+		return nil, err
+	}
+	logrus.Debugf("Got entry for key '%s': '%s'", entryKey, raw)
+
+	err = json.Unmarshal(raw, &entry)
+	if err != nil {
+		msg := fmt.Sprintf("Error unmarshalling JSON for entry '%s': %v  (json str: '%s')", id, err, raw)
+		logrus.Error(msg)
+		return nil, errors.Wrap(err, msg)
+	}
+
+	// now we interleave the visit count and the last visit time
+	// from the redis sources (we do this so we don't have to rewrite
+	// the entry every time someone visits which is madness)
+	//
+	// first, the visit count is just the length of the visitors list
+	entryVisitsKey := entryVisitsPrefix + id
+	visitCount, err := r.c.LLen(entryVisitsKey).Result()
+	if err != nil {
+		logrus.Warnf("Could not get length of visitor list for id '%s': '%v'", id, err)
+		entry.Public.VisitCount = int(0) // or zero if nobody's visited, that's fine.
+	} else {
+		entry.Public.VisitCount = int(visitCount)
+	}
+
+	// grab the timestamp out of the last visitor on the list
+	lastVisit = time.Time(time.Unix(0, 0)) // default to start-of-epoch if we can't figure it out
```

Comment:
I would simplify the Variable declarations to `:=` and remove the declaration in line 203

Reply:
done